pkg/util/strategicpatch/patch.go

```diff
@@ -160,12 +155,25 @@ func CreateTwoWayMergePatch(original, modified []byte, dataStruct interface{}, f
 		}
 	}
 
+	patchMap, err := CreateTwoWayMergeMapPatch(originalMap, modifiedMap, dataStruct, fns...)
+	if err != nil {
+		return nil, err
+	}
+
+	return json.Marshal(patchMap)
+}
+
+// CreateTwoWayMergeMapPatch creates a patch from an original and modified JSON objects,
+// encoded map[string]interface{} representation (the children in this representation
+// can be map[string]interface[], []interface{} or primitive type).
+// The serialized version of the map can then be passed to StrategicMergeMapPatch.
+func CreateTwoWayMergeMapPatch(original, modified map[string]interface{}, dataStruct interface{}, fns ...PreconditionFunc) (map[string]interface{}, error) {
```

Comment:
We should probably go ahead and create a type alias for `map[string]interface{}` in this package as `JSONMap` or something similar with a nice Godoc explaining what the type allows.